### ISSUE TITLE
SslError not handled during connectivity testing

### ISSUE THREAD

**Issue by domainabusers** [nabla-c0d3/sslyze#562]
**Describe the bug**
Unlike `_nassl.OpenSSLError`, `_nassl.SslError` is not handled in server_connectivity.py. Further, neither `OpenSSLError` nor `SslError` are handled during client certificate requirement testing. I believe this causes the scanner to hang indefinitely if the remote server closes the connection during connectivity testing ("Connection was shut down by peer").

**To Reproduce**
1. Install SSLyze using pip
2. Run the API sample code on a server which closes the connection during client certificate requirement testing.
3. See error

**Expected behavior**
The `SslError` in turn throws `ConnectionToServerFailed`, perhaps with an indication of which stage of connection testing the connection failed. I believe this will in turn add the terminated thread to the results queue in _mass_connectivity_tester.py.

**Python environment (please complete the following information):**
 - OS: CentOS 7
 - Python version: 3.8.11

**Additional context**
I believe `SslError` is also used for I/O errors in nassl. I queued a large number of servers in SSLyze and am thus unsure which server is causing the `SslError`. I have not tried to reproduce the error myself.

**Comment by domainabusers** [comment]
This issue appears to be triggered when testing for TLS 1.3 client certificates.

**Comment by nabla-c0d3** [comment]
@domainabusers Do you have a server I can test this on? Otherwise I will not be able to fix this issue. Thanks!

**Comment by domainabusers** [comment]
I've sent some examples by email.

**Comment by qfayet** [comment]
I experienced an error type that is not taken into account by this fix:

```
nassl._nassl.OpenSSLError: error:1408F119:SSL routines:ssl3_get_record:decryption failed or bad record mac
```

To reproduce:

```shell
python3 -m sslyze webservices.daf.com
```

Would it be possible to take it into account?

Maybe by adding this line in the module `tls_connection.py` :

```python
_HANDSHAKE_REJECTED_TLS_ERRORS = {
    ...
    "bad record mac": "TLS error: bad record mac",
    ...
}
```

**Comment by nabla-c0d3** [comment]
Fix released as part of [v5.0.6](https://github.com/nabla-c0d3/sslyze/releases/tag/5.0.6).
